### PR TITLE
Improving Combat Carrier Viability

### PR DESCRIPTION
**Pull request by Araulius**
## The TL;DR

Decreases hugger delay (the road block to the speed at which you can chuck huggers) from elder to primordial

## Why It's Good For The Game

With the advent of the new smartminigun, on top of an entire arsenal of automatic weapons which are compatible with aim mode, this PR attempts to make it easier to flood the combat zone with your full stockpile of a whole seven huggers (at ancient.)

This PR will also:

1. increase the usefulness of the "take aim" order.
2. increase the usefulness of swat masks
3. somewhat add a downside to adhering to the shotgun meta
4. somewhat counter(?) melee marine builds.
5. Increase the prevalence of larval jelly for medical staff. (this means lots more lemoline, and advanced combat chems.)
6. Increased role Importance of the smartgunner class.

In summary: This will make carrier more useful in a combat support roll at the higher maturities, while encouraging use of more unused marine tools.

## Changelog
hugger delay at elder reduced from 2.5 to 1.5
hugger delay at ancient reduced from 1.5 to 1.0
hugger delay at primordial reduced from 1.5 to 0.8

### CODE REVIEW

**Comment by Naaanii** [comment]
Nice buff (revert).

**Comment by FosterGrime-Heymynameslouie** [comment]
very cool cumbat carrier that will defitntely get obliterated by a sea of angry marines getting instantly pregnated with primo carriers.

what exactly does "the increase the usefulness of the "take aim" order." do though? I dont see any changes done to the orders part unless maybe im reading code wrong.

**Comment by Araulius** [comment]
> very cool cumbat carrier that will defitntely get obliterated by a sea of angry marines getting instantly pregnated with primo carriers.
> 
> what exactly does "the increase the usefulness of the "take aim" order." do though? I dont see any changes done to the orders part unless maybe im reading code wrong.

By providing them more scenarios in which they will be useful in.

**Comment by FosterGrime-Heymynameslouie** [comment]
> > very cool cumbat carrier that will defitntely get obliterated by a sea of angry marines getting instantly pregnated with primo carriers.
> > what exactly does "the increase the usefulness of the "take aim" order." do though? I dont see any changes done to the orders part unless maybe im reading code wrong.
> 
> By providing them more scenarios in which they will be useful in.

oh
I completely didint think of that.

based pr spede mrghe

**Comment by goose4429** [comment]
Anc and Primo delay is surely too low, I'd put them to like 1.2 and 1.0. Poor noobies getting raped by bunch of red huggers is not good, and 0.8 really allows you to spam them. 1.5 -> 0.8 is 2x speed increase.

**Comment by ilikesaying** [comment]
Woe

Huggers be upon ye

**Comment by Araulius** [comment]
> Anc and Primo delay is surely too low, I'd put them to like 1.2 and 1.0. Poor noobies getting raped by bunch of red huggers is not good, and 0.8 really allows you to spam them. 1.5 -> 0.8 is 2x speed increase.

Poor noobies able to pump out 30 rounds with a bog standard assault rifle in the time it takes to throw seven huggers. Woe to the marines with an infinite ammo 1000 round capacity minigun. Won't SOMEONE THINK OF THE MARINES? Marines have the tools to deal with this, easily. Not using them should rightly be punished.

**Comment by Araulius** [comment]
Ways a single marine in a small group can completely shut down hugger floods:

1. Aim mode with service pistol
2. bipod aim mode t-60
3. literally any marksman rifle with a miniscope
4. whack them with Literally any tool (including the stock of your gun)
6. smart machinegun
7. smart MINIGUN (BRRRRRRRRRRT)
8. Fire
9. take aim command
10. swat masks
11. smg-90 (large mag capacity, fast fire rate)
12. ppsh aim mode
13. t-12 aim mode
14. t-18 aim mode
15. literally the rest of the marine arsenal on aim mode in the hands of a single marine

All it takes is for one dude to  focus the huggers. If marines can't handle that, that's called a **skill issue.**

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/castedatum_carrier.dm [hunk 1]
```diff
@@ -191,7 +191,7 @@
 
 	// *** Carrier Abilities *** //
 	huggers_max = 7
-	hugger_delay = 1.5 SECONDS
+	hugger_delay = 0.8 SECONDS
```
Comment (DeltaFire15) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/castedatum_carrier.dm:194]: primordial stat-wise is supposed to be the same as ancient according to maints

**Comment by Losenis** [comment]
I'd be in favor of this if spamming huggers onto a tile didn't need one hugger to die as you pixelhunt it, click it, wait for ping, wait for bullet travel time, wait for the next hugger to be clickable, repeat, before you can actually target the rest of the huggers. Clawed ones are the absolute worst to deal with because they deliberatedly all jump onto the same tile every now and then (IE, onto a marine). Smartminigun is also pretty shit at shutting huggers down because it also adds the windup onto the equation.

**Review by Pariah919 — COMMENTED** [review]
No.

**Comment by Araulius** [comment]
> No.

**Comment by Araulius** [comment]
> I'd be in favor of this if spamming huggers onto a tile didn't need one hugger to die as you pixelhunt it, click it, wait for ping, wait for bullet travel time, wait for the next hugger to be clickable, repeat, before you can actually target the rest of the huggers. Clawed ones are the absolute worst to deal with because they deliberatedly all jump onto the same tile every now and then (IE, onto a marine). Smartminigun is also pretty shit at shutting huggers down because it also adds the windup onto the equation.
> 
> Also this all assumes an ideal scenario where the carrier is alone and the huggers are not actually shielded by the rest of the xenos blocking IFF shots before the huggers jump through said xenos onto marines. The rounds where they get like 30 people hugged and even some bursting should indicate as to what happens in practice. This'd just allow carrier to get people hugged in instances where it really shouldn't, because only one larval hugger needs to actually get through to land an embryo, and it'll also be exceptionally difficult to remove it from him when there's half a dozen more hughers around with a few xenos to block IFF shots towards them as they fight.
> 
> Overall not really seeing the logic in this buff, primo carrier already has hugger turret on top of that. Carrier is already strong enough if played correctly, and pretty unfun to deal with as it

**Comment by Al-1ce** [comment]
The complications that will cause is an entire can of worms on its own;

>increase the usefulness of the "take aim" order


This is supposed to make aiming instant. As far as I've tested, that isn't the case. It appears to be broken


>increase the usefulness of swat masks

Sure. Except, the usefulness of a single swat mask is short lived when the huggers you're _wearing the mask to avoid_ can latch on much quicker than you can react, mask or not.


>somewhat add a downside to adhering to the shotgun meta


This is an understatement to say the least. The delay between T-29 shots (and T-35 in some instances) will always be slower tha n how long it takes for a hugger to pounce. You are not considering just how much this affects the overall state of balance. 


>somewhat counter(?) melee marine builds


Need I say it again? You either don't seem to care or consider that this change is heavily affecting balance. It becomes more than a tiny stat change. You do not see PRs getting merged here that make three or more marine strategies weaker / inefficient for a reason. Melee especially suffers from Carrier as getting in close (a requirement to land a hit) always results in the activation time of a hugger being much faster.


>Increase the prevalence of larval jelly for medical staff. (this means lots more lemoline, and advanced combat chems.)


You entirely avoid mentioning that this renders a sizable amount of marines stuck shipside for however long it takes for their larva to be removed. Additional lemoline is rarely used and advanced combat chems are never made but that's more of a player issue I suppose.


>Increased role Importance of the smartgunner class.


Being a dedicated anti-hugger machine isn't really suit for the job. Granted, they can do it well, but so can any other marine with an automatic rifle. All this does is increase the ammo consumption of SGs and let them waste their time on useless targets.


As I mentioned in the start of this response, can of worms. You're making a change that fucks with way too many other variables and ultimately ends up as a nightmare for balance.

No.

**Comment by BraveMole** [comment]
Mjp said no. @Pariah919 Could you elaborate on why you don't like next time ?

**Comment by Naaanii** [comment]
Increasing the cd at which huggers can be generated as well as plasma might help this notion. While hugger spam can be annoying to deal with, increased cds and plasma costs ensure more downtime between epic carrier rapid fire moments. Traps right now already eat up most of a carrier's plasma hence the more combat oriented playstyle is more favourable for most players.

A good carrier can overcome said suggestions by placing eggs, but again that includes more prep time, can only be done a few times until marines gain more ground and can barely be used to save time and make hugger spam completely viable. 0.8 seconds is a bit too low and the turret is good enough for primo carrier imo and like Losenis mentioned, live hugger sprites should come on top of dead ones so pixel hunting isn't a norm.

Fire still does counter huggers quite well enough and a marine on fire prevents them from even jumping or targetting said marine is a thing to note. Carriers aren't really a facetank caste and can easily be punished for overstepping and while they can be annoying to deal with, spamming combat huggers can have a downside to itself.
